src/controllers/users.controller.js

```
@@ -0,0 +1,103 @@
+const usersService = require('../repository/users.repository');
+
+const get = async (_req, res) => {
+  try {
+    const users = await usersService.getAll();
+
+    res.json(users.map((user) => usersService.normilizeUser(user)));
+  } catch (error) {
+    res.sendStatus(500);
+  }
+};
+
+const create = async (req, res) => {
+  const { name } = req.body;
+
+  if (!name) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  try {
+    const user = await usersService.create(name);
+
+    res.status(201).json(usersService.normilizeUser(user));
+  } catch (error) {
+    res.sendStatus(500);
+  }
+};
+
+const getOne = async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const user = await usersService.getById(id);
+
+    if (!user) {
+      res.sendStatus(404);
+
+      return;
+    }
+    res.json(usersService.normilizeUser(user));
+  } catch (error) {
+    res.sendStatus(500);
+  }
+};
+
+const remove = async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const todo = await usersService.getById(id);
+
+    if (!todo) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    await usersService.remove(id);
+
+    res.sendStatus(204);
+  } catch (error) {
+    res.sendStatus(500);
+  }
+};
+
+const update = async (req, res) => {
+  const { id } = req.params;
+  const { name } = req.body;
+
+  if (typeof name !== 'string') {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  try {
+    const user = await usersService.getById(id);
+
+    if (!user) {
+      res.sendStatus(404);
+
+      return;
+    }
+
+    await usersService.update({ id, name });
+
+    const updatedUser = await usersService.getById(id);
+
+    res.json(usersService.normilizeUser(updatedUser));
+  } catch (error) {
+    res.sendStatus(500);
```

Catch blocks return `500` but swallow the error. For easier debugging during tests and development, log the error before sending the response. For example:
```
} catch (error) {
  console.error(error);
  res.sendStatus(500);
}
```
Apply this to the catch blocks (e.g., lines 4–9, 22–27, 34–44, 51–64, 78–93). This doesn't change behavior for clients but helps investigators see the underlying cause when tests or DB ops fail.